agent/consul/leader_connect_ca.go

```diff
@@ -564,17 +561,13 @@ func (c *CAManager) primaryInitialize(provider ca.Provider, conf *structs.CAConf
 	}
 
 	// Store the root cert in raft
-	resp, err := c.delegate.ApplyCARequest(&structs.CARequest{
+	_, err = c.delegate.ApplyCARequest(&structs.CARequest{
 		Op:    structs.CAOpSetRoots,
 		Index: idx,
 		Roots: []*structs.CARoot{rootCA},
 	})
 	if err != nil {
-		c.logger.Error("Raft apply failed", "error", err)
-		return err
-	}
-	if respErr, ok := resp.(error); ok {
-		return respErr
+		return fmt.Errorf("raft apply failed: %w", err)
```

Comment:
In general the pattern of "log the error, then return the error" is a problem. Errors should only be logged from where they are handled. The problem here is that we end up with separate log lines, that either makes it look like this event has happened twice, or otherwise repeats information at potentially different log levels. It makes it difficult to associate the "raft apply failed" message with the operation that failed (CA initialization).

I've fixed this in other places, so I'm doing here along with the other cleanup. The caller of this function will handle the error, and the full error with wrapped context is logged where it is handled.